rusty-machine/src/linalg/utils.rs

```diff
@@ -111,6 +112,64 @@ pub fn unrolled_sum<T>(mut xs: &[T]) -> T
     sum
 }
 
+/// Vectorized binary operation applied to two slices.
+/// The first argument should be a mutable slice which will
+/// be modified in place to prevent new memory allocation.
+///
+/// # Examples
+///
+/// ```
+/// use rusty_machine::linalg::utils;
+///
+/// let mut a = vec![2.0; 10];
+/// let b = vec![3.0; 10];
+///
+/// utils::in_place_vec_bin_op(&mut a, &b, |x, &y| { *x = 1f64 + *x * y });
+///
+/// // Will print a vector of `7`s.
+/// println!("{:?}", a);
+/// ```
+pub fn in_place_vec_bin_op<F, T: Copy>(mut u: &mut [T], v: &[T], mut f: F)
+    where F: FnMut(&mut T, &T) {
+        debug_assert_eq!(u.len(), v.len());
+        let len = cmp::min(u.len(), v.len());
+        
+        let ys = &v[..len];
+        
+        let mut new_slice = mem::replace(&mut u, &mut []);
+        let (l,_) = new_slice.split_at_mut(len);
+        u = l;
```

Comment:
This can be simplified to just `let xs = &mut u[..len]` (and adjust the use of `u` down the code).